music_metadata_extractor/models.py

```diff
@@ -102,14 +103,17 @@ def __repr__(self):
 class DictInput(BaseProviderInput):
     """Provider input class with a dict that contains song and artist name"""
 
-    def __init__(self, song_name: str, artist_name: str):
+    def __init__(self, song_name: str, artist_name: str, song_ytviews: int, song_ytdate: dt):
```

Comment:
Why are these attributes needed for the provider input? Provider input is expected to contain details that the provider (which in our case, is only one and is Spotify) would need to fetch data.

So ideally, rather than add anything here at all, it might be better to rename [`get_provider_input`](https://github.com/lttkgp/metadata-extractor/blob/8305912de50faa39b555c2cd511c1060502ee562/music_metadata_extractor/resolver.py#L32) to something like `get_source_data` and add an extra return value named `extraAttrs` of type `Dict` (so we don't need to pre-define a class and every input source can have different extra attributes to return). And add these details there.

So, `get_yt_info` will be returning this as a second return value that might look like:
```py
{
  "youtube": {
    "views": views
    "posted_date": posted_date
  }
}
```
In future too, if we want to return extra attributes, we can just add it here.

Reply:
Yes, that's a great idea.